### DISCUSSION TITLE
How to Save Logs to Local Files

### DISCUSSION THREAD

**Discussion by Hbbbit** [eclipse-theia/theia#13998] [Q&A]
Hey everyone,

I'm currently working with Theia and I'm trying to figure out how to save logs directly to a local file. I've been using Theia for a while now, but I'm not sure what the best approach is for this specific task.

I've checked the documentation and searched through the community forums, but I couldn't find a clear answer. Could someone please guide me through the steps or point me to the right resources?

Here are a few specific questions I have:

What configuration changes are necessary to enable logging to a file?
Are there any specific plugins or extensions that facilitate log file creation in Theia?
How can I specify the log file path and format?
Any help or pointers would be greatly appreciated. Thanks in advance!

**Comment by msujew** [comment]
Hey @Huzhibo12138,

The IDE/framework doesn't support logging to a file directly. If you want to contribute such a functionality to the framework, please take a look at [this file](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/node/logger-cli-contribution.ts).

Generally, you can always run any Theia based app in a way that all logged output is written to a file via the `> log.txt` (on Linux systems) or `> log.txt 2>&1` (on Windows) terminal arguments.
